File: problemas/iniciante/1115/1115.cpp
#include <iostream>

using namespace std;

int main()
{
    int X, Y;

    while (cin >> X >> Y)
    {
        if (X == 0 || Y == 0)
        {
            break;
        }

        if (X > 0)
        {
            if (Y > 0)
            {
                cout << "primeiro" << endl;
            }
            else
            {
                cout << "quarto" << endl;
            }
        }
        else
        {
            if (Y > 0)
            {
                cout << "segundo" << endl;
            }
            else
            {
                cout << "terceiro" << endl;
            }
        }
    }

    return 0;
}

File: problemas/iniciante/1115/1115.hs
import Text.Printf

imprime :: Int -> Int -> IO ()
imprime 0 y = return ()
imprime x 0 = return ()
imprime x y = do
  if x > 0
    then
      if y > 0
        then printf "primeiro\n"
        else printf "quarto\n"
    else
      if y > 0
        then printf "segundo\n"
        else printf "terceiro\n"
  entrada <- getLine
  let [x, y] = map read (words entrada) :: [Int]
  imprime x y

main :: IO ()
main = do
  entrada <- getLine
  let [x, y] = map read (words entrada) :: [Int]
  imprime x y
